web/src/reducers/dataReducerForHistoryFeature.js

```diff
@@ -0,0 +1,226 @@
+  zone.timezone = zoneConfig.timezone;
+  zone.shortname = translation.getZoneNameWithCountry(key);
+  zone.hasParser = (zoneConfig.parsers || {}).production !== undefined;
+  zone.hasData = zone.hasParser;
+  zone.delays = zoneConfig.delays;
+  zone.disclaimer = zoneConfig.disclaimer;
+});
+// Add id to each zone
+Object.keys(zones).forEach((k) => { zones[k].countryCode = k; });
+
+// ** Prepare initial exchange data
+const exchanges = Object.assign({}, exchangesConfig);
+Object.entries(exchanges).forEach((entry) => {
+  const [key, value] = entry;
+  value.countryCodes = key.split('->').sort();
+  if (key.split('->')[0] !== value.countryCodes[0]) {
+    console.warn(`Exchange sorted key pair ${key} is not sorted alphabetically`);
+  }
+});
+
+const initialDataState = {
+  // Here we will store data items
+  grid: { zones, exchanges },
+  hasConnectionWarning: false,
+  hasInitializedGrid: false,
+  histories: {},
+  isLoadingHistories: false,
+  isLoadingGrid: false,
+  isLoadingSolar: false,
+  isLoadingWind: false,
+  solar: null,
+  wind: null,
+  solarDataError: null,
+  windDataError: null,
+};
+
+const reducer = (state = initialDataState, action) => {
+  switch (action.type) {
+    case 'GRID_DATA_FETCH_REQUESTED': {
+      return { ...state, hasConnectionWarning: false, isLoadingGrid: true };
+    }
+
+    case 'GRID_DATA_FETCH_SUCCEEDED': {
+      // Create new grid object
+      const newGrid = Object.assign({}, {
+        zones: Object.assign({}, state.grid.zones),
+        exchanges: Object.assign({}, state.grid.exchanges),
+      });
+      // Create new state
+      const newState = Object.assign({}, state);
+      newState.grid = newGrid;
+
+      // Reset histories that expired
+      newState.histories = Object.assign({}, state.histories);
+      Object.keys(state.histories).forEach((k) => {
+        const history = state.histories[k];
+        const lastHistoryMoment = new Date(history.at(-1).stateDatetime);
+        const stateMoment = new Date(action.payload.datetime);
+        if (addMinutes(lastHistoryMoment, 15) < stateMoment) {
+          delete newState.histories[k];
+        }
+      });
+
+      // Set date
+      newGrid.datetime = action.payload.datetime;
+
+      // Reset all data we want to update (for instance, not maxCapacity)
+      Object.keys(newGrid.zones).forEach((key) => {
+        const zone = Object.assign({}, newGrid.zones[key]);
+        zone.co2intensity = undefined;
+        zone.fossilFuelRatio = undefined;
+        zone.fossilFuelRatioProduction = undefined;
+        zone.renewableRatio = undefined;
+        zone.renewableRatioProduction = undefined;
+        zone.exchange = {};
+        zone.production = {};
+        zone.productionCo2Intensities = {};
+        zone.productionCo2IntensitySources = {};
+        zone.dischargeCo2Intensities = {};
+        zone.dischargeCo2IntensitySources = {};
+        zone.storage = {};
+        zone.source = undefined;
```

Comment:
Here we could only reset what we're getting in state, eg zoneStateOverview

   {
            "countryCode": zone_key,
            "stateDatetime": format_utc_datetime(state_datetime),
            "estimationMethod": estimation_method,
            "co2intensity": data["carbon_intensity_avg"],
            "co2intensityProduction": data["carbon_intensity_production_avg"],
            "fossilFuelRatio": percentage_to_ratio(
                data, "power_origin_percent_fossil_avg"
            ),
            "renewableRatio": percentage_to_ratio(
                data, "power_origin_percent_renewable_avg"
            ),
            "fossilFuelRatioProduction": percentage_to_ratio(
                data, "power_production_percent_fossil_avg"
            ),
            "renewableRatioProduction": percentage_to_ratio(
                data, "power_production_percent_renewable_avg"
            ),
        }